### ISSUE TITLE
Installation issue: gmake requires awk

### ISSUE THREAD

**Issue by michaelkuhn** [LLNL/spack#41208]
### Steps to reproduce the issue

Trying to install any autotools package in a minimal OpenSUSE Tumbleweed container fails because gmake requires awk. Adding a dependency is not possible since that introduces a circular dependency.

```console
$ spack spec -I gmake
Input spec
--------------------------------
 -   gmake

Concretized
--------------------------------
 -   gmake@4.4.1%gcc@13.2.1~guile build_system=generic arch=linux-opensusetumbleweed2.38-zen2
```

### Error message

<details><summary>Error message</summary>

<pre>
1 error found in build log:
     208    config.status: creating doc/Makefile
     209    ./config.status: line 1234: awk: command not found
     210    config.status: creating po/Makefile.in
     211    ./config.status: line 1234: awk: command not found
     212    config.status: creating src/config.h
     213    ./config.status: line 1261: awk: command not found
  >> 214    config.status: error: could not create src/config.h
</pre></details>

### Information on your system

* **Spack:** 0.22.0.dev0 (f2192a48cebbbe926e68bb956e4613626c9d5f32)
* **Python:** 3.11.5
* **Platform:** linux-opensusetumbleweed2.38-zen2
* **Concretizer:** clingo

### Additional information

@haampie

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by haampie** [comment]
I think awk is supposed to be part of any minimal distro image (edit: [posix compliant](https://pubs.opengroup.org/onlinepubs/9699919799/idx/utilities.html) I mean)

**Comment by michaelkuhn** [comment]
Should we then list this as part of the system prerequisites? I guess it's a non-issue for most images (I just checked Fedora, Ubuntu and Alpine, which include some implementation), but at least the Tumbleweed container is missing it.

**Comment by haampie** [comment]
Yeah, I think we could add a subset of the list of posix utilities.

Personally I think the full list is too big: includes stuff like m4 which we definitely add as build dep, but also `file`, which autotools configure scripts usually hard-code at `/usr/bin/file` even though it's a trivial build dep to add.
